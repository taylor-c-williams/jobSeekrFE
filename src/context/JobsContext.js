import { createContext, useContext, useMemo, useState, useEffect } from 'react'
import { getAllUserJobs } from '../services/users'

const JobsContext = createContext()

const JobsProvider = ({ children }) => {
  const [jobs, setJobs] = useState([])

  useEffect(() => {
    const fetchJobs = async () => {
      try {
        const currentJobs = await getAllUserJobs()
        setJobs(currentJobs)
      } catch (e) {
        console.log(e.message)
      }
    }
    fetchJobs()
  }, [])

  const value = useMemo(() => ({ jobs, setJobs }), [jobs])
  return <JobsContext.Provider value={value}> {children} </JobsContext.Provider>
}

const useJobs = () => {
  const context = useContext(JobsContext)
  if (context === undefined) {
    throw new Error('useJobs must be used within a JobsProvider!')
  }
  return context
}

export { JobsContext, JobsProvider, useJobs }
