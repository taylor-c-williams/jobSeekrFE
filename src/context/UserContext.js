import { createContext, useContext, useMemo, useState, useEffect } from 'react'
import { getUser } from '../services/users'

const UserContext = createContext()

const UserProvider = ({ children }) => {
  const [user, setUser] = useState({})

  useEffect(() => {
    const fetchCurrentUser = async () => {
      try {
        const currentUser = await getUser()
        if (currentUser.username) {
          setUser({
            ...currentUser,
          })
        } else {
          setUser({})
        }
      } catch (e) {
        console.log(e.message)
      }
    }
    fetchCurrentUser()
  }, [])

  const value = useMemo(() => ({ user, setUser }), [user])
  return <UserContext.Provider value={value}> {children} </UserContext.Provider>
}

const useUser = () => {
  const context = useContext(UserContext)
  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider!')
  }
  return context
}

export { UserContext, UserProvider, useUser }
