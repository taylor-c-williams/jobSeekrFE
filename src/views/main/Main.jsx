import { useState } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import LogIn from '../../components/logIn/LogIn'
import SignUp from '../../components/signUp/SignUp'
import { useUser } from '../../context/UserContext'
import { getUser } from '../../services/users'
import { logIn, signUp } from '../../services/auth'
import styles from './Main.module.css'

export default function Main({ login }) {
  const [inputData, setInputData] = useState({ username: '', password: '' })
  const { setUser } = useUser()
  const navigate = useNavigate()

  const handleInputData = (e) => {
    const { value, name } = e.target
    setInputData({ ...inputData, [name]: value })
  }

  const handleSignUp = async () => {
    try {
      const { username, password } = inputData
      await signUp(username, password)
      await logIn(username, password)
      const newUser = await getUser()
      await setUser(newUser)
      navigate('/home')
    } catch (error) {
      console.error('signup error')
    }
  }

  return (
    <div>
      <h1>seeker</h1>
      {login ? (
        <div>
          <LogIn />
          <section className={styles.signUpLink}>
            Don't have an account? <Link to='/signup'>Sign up</Link>
          </section>
        </div>
      ) : (
        <div>
          <SignUp
            handleSubmit={handleSignUp}
            handleInputData={handleInputData}
            inputData={inputData}
          />
          <section className={styles.signUpLink}>
            Already have an account? <Link to='/'>Log In</Link>
          </section>
        </div>
      )}
    </div>
  )
}
