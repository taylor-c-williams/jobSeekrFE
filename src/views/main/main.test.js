import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { UserProvider } from '../../context/UserContext'
import Main from './Main'

it.skip('renders the Main view & login component accurately', async () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <Main login={true} />
      </UserProvider>
    </MemoryRouter>
  )
  const text = await screen.findByText("Don't have an account?")
  expect(container).toMatchSnapshot()
  expect(text).toBeInTheDocument()
})

it.skip('renders the Main view & sign up component accurately', async () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <Main login={false} />
      </UserProvider>
    </MemoryRouter>
  )
  const text = await screen.findByText('Already have an account?')
  expect(container).toMatchSnapshot()
  expect(text).toBeInTheDocument()
})
