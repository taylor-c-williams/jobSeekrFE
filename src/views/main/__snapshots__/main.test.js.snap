// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Main view & login component accurately 1`] = `
<div>
   
  <div>
    <h1>
      seeker
    </h1>
    <div>
      <div
        class="main"
      >
        <fieldset>
          <legend>
            Log In
          </legend>
          <label>
            <section
              class="labelTitle"
            >
              <svg
                aria-hidden="true"
                class="labelIcon"
                fill="none"
                height="15"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.121 17.804A13.937 13.937 0 0112 16c2.5 0 4.847.655 6.879 1.804M15 10a3 3 0 11-6 0 3 3 0 016 0zm6 2a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              Username
            </section>
            <input
              label="username"
              spellcheck="false"
            />
          </label>
          <label>
            <section
              class="labelTitle"
            >
              <svg
                aria-hidden="true"
                class="labelIcon"
                fill="none"
                height="15"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 11c0 3.517-1.009 6.799-2.753 9.571m-3.44-2.04l.054-.09A13.916 13.916 0 008 11a4 4 0 118 0c0 1.017-.07 2.019-.203 3m-2.118 6.844A21.88 21.88 0 0015.171 17m3.839 1.132c.645-2.266.99-4.659.99-7.132A8 8 0 008 4.07M3 15.364c.64-1.319 1-2.8 1-4.364 0-1.457.39-2.823 1.07-4"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              Password
            </section>
            <input
              spellcheck="false"
              type="password"
            />
          </label>
          <button>
            Log In
          </button>
        </fieldset>
      </div>
      <section
        class="signUpLink"
      >
        Don't have an account? 
        <a
          href="/signup"
        >
          Sign up
        </a>
      </section>
    </div>
  </div>
   
</div>
`;

exports[`renders the Main view & sign up component accurately 1`] = `
<div>
   
  <div>
    <h1>
      seeker
    </h1>
    <div>
      <div
        class="main"
      >
        <fieldset>
          <legend>
            Sign Up
          </legend>
          <label>
            <section
              class="labelTitle"
            >
              <svg
                aria-hidden="true"
                class="labelIcon"
                fill="none"
                height="15"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.121 17.804A13.937 13.937 0 0112 16c2.5 0 4.847.655 6.879 1.804M15 10a3 3 0 11-6 0 3 3 0 016 0zm6 2a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              Username
            </section>
            <input
              label="username"
              name="username"
              spellcheck="false"
              value=""
            />
          </label>
          <label>
            <section
              class="labelTitle"
            >
              <svg
                aria-hidden="true"
                class="labelIcon"
                fill="none"
                height="15"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 11c0 3.517-1.009 6.799-2.753 9.571m-3.44-2.04l.054-.09A13.916 13.916 0 008 11a4 4 0 118 0c0 1.017-.07 2.019-.203 3m-2.118 6.844A21.88 21.88 0 0015.171 17m3.839 1.132c.645-2.266.99-4.659.99-7.132A8 8 0 008 4.07M3 15.364c.64-1.319 1-2.8 1-4.364 0-1.457.39-2.823 1.07-4"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              Password
            </section>
            <input
              label="password"
              name="password"
              spellcheck="false"
              type="password"
              value=""
            />
          </label>
          <button>
            Sign Up
          </button>
        </fieldset>
      </div>
      <section
        class="signUpLink"
      >
        Already have an account? 
        <a
          href="/"
        >
          Log In
        </a>
      </section>
    </div>
  </div>
   
</div>
`;
